### PR TITLE
Significantly simpler and malleable test setup

### PR DESCRIPTION
**Pull request by ashwinb**
# What does this PR do?

Significantly simplifies running tests. 

Previously you ran tests by doing:
```bash
MODEL_ID=<...> 
PROVIDER_ID=<...> 
PROVIDER_CONFIG=config.yaml pytest -s llama_stack/providers/tests/inference/test_inference.py
```
This was pretty annoying because 
- you needed to change YAMLs _and_ CLI parameters
- one needed to explicitly document how these environment variables affected, and especially how they affected tests where providers for multiple APIs needed to be used
- the fixtures weren't quite reusable, we weren't using pytest properly

This PR fixes these issues. If you wanted to run tests with Fireworks you simply need to:
```bash
pytest -s -v -m "fireworks" llama_stack/providers/tests/inference/test_inference.py \
  --env FIREWORKS_API_KEY=<...>
```

You can easily limit to running specific providers / models
```bash
# only run 3b model for ollama
pytest -s -v -m "fireworks or (ollama and llama_3b)" llama_stack/providers/tests/inference/test_inference.py \
  --env FIREWORKS_API_KEY=<...>
```

### CODE REVIEW

File: llama_stack/providers/tests/inference/conftest.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        provider_type="remote::ollama",
+        config=(
+            OllamaImplConfig(
+                host="localhost", port=os.getenv("OLLAMA_PORT", 11434)
+            ).model_dump()
+        ),
+    )
+
+
+@pytest.fixture(scope="session")
```
Comment (ashwinb) [inline llama_stack/providers/tests/inference/conftest.py:65]: these fixtures are now completely reusable. we can formalize this env-var -> config pattern also and push it into how we run things with docker

**Review by raghotham — APPROVED** [review]
Looks good thanks. Will be good to add docs on how to add new tests

**Comment by ashwinb** [comment]
A lot more is coming here to make sure Agents tests are as simple as possible. And yes, I won't land until docs are added this time.

**Comment by yanxi0830** [comment]
https://github.com/meta-llama/llama-stack-apps/pull/106/files#diff-8d76e49de7c5d0a3300be753b9346e1b6fd535c394b1d9a2c4bd798cfd407193 

I revamped the agents test using client SDK in llama-stack-apps here. Maybe these can be moved to this repo or llama-stack-client-python repo?

**Pull request merged**